src/osd/SnapMapper.cc

```diff
@@ -208,10 +208,13 @@ int OSDriver::get_next_or_current(
 string SnapMapper::get_prefix(int64_t pool, snapid_t snap)
 {
   static_assert(sizeof(pool) == 8, "assumed by the formatting code");
+
+  // note: the snap_id is to be formatted as a 64-bit hex number,
+  // and not according to the text representation of snapid_t
```

Comment:
If we create snapmapper entries _only_ for clone objects (which I think is true), the faulty formatting shouldn't have any real impact and the fix should be ok. 

It looks like `OSDMonitor` also uses the same casting, so that might help to rule out other objects rather than clone ones:
```
string OSDMonitor::make_purged_snap_key(int64_t pool, snapid_t snap)
{
  char k[80];
  snprintf(k, sizeof(k), "purged_snap_%llu_%016llx",
	   (unsigned long long)pool, (unsigned long long)snap);
  return k;
}
```

Perhaps we should assert that `snap != NOSNAP && snap != SNAPDIR` before/while formatting.

Reply:
The assert() suggestion makes a lot of sense.
I'll try to verify the other points.
Thanks